docs/vtpm.md

```diff
@@ -0,0 +1,41 @@
+```ts
+ vTPM Object
+ {
+  id: string,
+  is_protected: boolean, // Currently unused by the XAPI, may be used in future.
+  is_unique: boolean // Currently unused by the XAPI, may be used in future.
+ }
+```
+
+- `is_protected`: The TPM is encrypted whenever it's transferred or at stored to disk (and exported)
+- `is_unique`: The TPM contents are exclusive to that VM, that means it has never been cloned (and maybe never exported nor imported)
+
+vTPM features are only enabled if XAPI is recent enough (check for `pool.restrict` or version (>= 8.3)).
+
+- vTPM
+  - At VM creation
+    - A toggle will be displayed in advanced settings under `boot firmware`.
+    - The default value is `false`, except for Windows >= W11.
```

Comment:
The exact check is not on the version of the OS, but on the contents of the template. We now know what exactly defines the need for vTPM in a template. See https://github.com/xenserver/guest-templates-json/blob/master/json/windows-11.json.

Extract:
```
    "platform": {
        "acpi_laptop_slate": "1",
        "vtpm": "true" <------- THIS
    },
```